flow/src/main/java/com/sequenceiq/flow/core/Flow2Handler.java

```diff
@@ -562,7 +549,28 @@ private void logFlowId(String flowId) {
         LOGGER.debug("Flow has been created with id: '{}' and the related request id: '{}'.", flowId, requestId);
     }
 
-    public FlowLog getFirstStateLogfromLatestFlow(Long resourceId) {
+    public FlowLog getFirstRetryableStateLogfromLatestFlow(Long resourceId) {
+        return FlowLogUtil.getFirstStateLog(findAllForLastFlowIdChecked(resourceId)).get();
+    }
+
+    public List<FlowLog> findAllForLastFlowIdChecked(Long resourceId) {
+        List<FlowLog> flowLogs = flowLogService.findAllForLastFlowIdByResourceIdOrderByCreatedDesc(resourceId);
+        Optional<FlowLog> pendingFlowLog = FlowLogUtil.getPendingFlowLog(flowLogs);
```

Comment:
As I see the code, you can filter the pending flow logs with SQL (JPQL)

Reply:
I would not delay this important fix for 2.52, but definitely worth to create a ticket for it and fix it in master, since this is an existing logic

Reply:
ok, @yg-gd @horadla23 could you please create a follow up for this?

Reply:
https://jira.cloudera.com/browse/CB-16161